### PR TITLE
fix: handle InfluxDBError when writing data

### PR DESCRIPTION
**Pull request by matthewelwell**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [ ] I have added information to `docs/` if required so people know about the feature!
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

Catch InfluxDBError when writing errors so we don't spam sentry for e.g. rate limiting errors. 

## How did you test this code?

Added a unit test. Note that the test is pretty crude for now as the caplog fixture is poor, but it achieves what we need.

### CODE REVIEW

File: api/app_analytics/influxdb_wrapper.py [hunk 1]
```diff
@@ -61,8 +62,8 @@ def add_data_point(self, field_name, field_value, tags=None):
     def write(self):
         try:
             self.write_api.write(bucket=settings.INFLUXDB_BUCKET, record=self.records)
-        except HTTPError:
-            logger.warning("Failed to write records to Influx.")
+        except (HTTPError, InfluxDBError) as e:
+            logger.warning("Failed to write records to Influx: %s", str(e))
```
Comment (khvn26) [inline api/app_analytics/influxdb_wrapper.py]: How about 

```suggestion
            logger.warning(
                "Failed to write records to Influx: %s",
                str(e),
                exc_info=e,
            )
```
?

Comment (matthewelwell) [reply api/app_analytics/influxdb_wrapper.py]: Oh nice, yeah, I didn't know about that 👍

File: api/tests/unit/app_analytics/test_unit_app_analytics_influxdb_wrapper.py [hunk 2]
```diff
@@ -24,15 +30,42 @@
...
+
 
+@pytest.fixture()
+def mock_write_api(mock_influxdb_client: MagicMock) -> MagicMock:
     mock_write_api = mock.MagicMock()
     mock_influxdb_client.write_api.return_value = mock_write_api
+    return mock_write_api
+
+
+def test_write(mock_influxdb_client: MagicMock, mock_write_api: MagicMock):
```
Comment (zachaysan) [inline api/tests/unit/app_analytics/test_unit_app_analytics_influxdb_wrapper.py]: Nit:
```suggestion
def test_write(mock_write_api: MagicMock) -> None:
```

**Review by zachaysan — APPROVED** [review]
Overall looks good. One small nit.

**Pull request merged**